server/sonar-server/src/main/java/org/sonar/server/webhook/ws/SearchAction.java

```diff
@@ -19,52 +19,54 @@
  */
 package org.sonar.server.webhook.ws;
 
-import com.google.common.base.Optional;
-import com.google.common.collect.ImmutableSet;
 import com.google.common.io.Resources;
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.List;
+import java.util.Optional;
 import javax.annotation.Nullable;
 import org.sonar.api.server.ws.Request;
 import org.sonar.api.server.ws.Response;
 import org.sonar.api.server.ws.WebService;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
 import org.sonar.db.component.ComponentDto;
-import org.sonar.server.setting.ws.Setting;
-import org.sonar.server.setting.ws.SettingsFinder;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.webhook.WebhookDto;
+import org.sonar.server.organization.DefaultOrganizationProvider;
 import org.sonar.server.user.UserSession;
 import org.sonarqube.ws.Webhooks.SearchWsResponse.Builder;
 
+import static java.util.Optional.ofNullable;
 import static org.apache.commons.lang.StringUtils.isNotBlank;
-import static org.sonar.api.web.UserRole.ADMIN;
 import static org.sonar.server.webhook.ws.WebhooksWsParameters.ORGANIZATION_KEY_PARAM;
 import static org.sonar.server.webhook.ws.WebhooksWsParameters.PROJECT_KEY_PARAM;
 import static org.sonar.server.webhook.ws.WebhooksWsParameters.SEARCH_ACTION;
 import static org.sonar.server.ws.KeyExamples.KEY_ORG_EXAMPLE_001;
 import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
 import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
+import static org.sonar.server.ws.WsUtils.checkStateWithOptional;
 import static org.sonar.server.ws.WsUtils.writeProtobuf;
 import static org.sonarqube.ws.Webhooks.SearchWsResponse.newBuilder;
 
 public class SearchAction implements WebhooksWsAction {
 
   private final DbClient dbClient;
   private final UserSession userSession;
-  private final SettingsFinder settingsFinder;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+  private final WebhookSupport webhookSupport;
 
-  public SearchAction(DbClient dbClient, UserSession userSession, SettingsFinder settingsFinder) {
+  public SearchAction(DbClient dbClient, UserSession userSession, DefaultOrganizationProvider defaultOrganizationProvider, WebhookSupport webhookSupport) {
     this.dbClient = dbClient;
     this.userSession = userSession;
-    this.settingsFinder = settingsFinder;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+    this.webhookSupport = webhookSupport;
   }
 
   @Override
   public void define(WebService.NewController controller) {
 
     WebService.NewAction action = controller.createAction(SEARCH_ACTION)
```

Comment:
Sorry for the late feedback. It would be better to rename api/webhooks/list.

Reply:
done